src/daemon/properties/mod.rs

```diff
@@ -146,32 +179,37 @@ impl Aggregate for Properties {
     type Command = PropertiesCommand;
     type StorableCommandDetails = StorablePropertiesCommand;
     type Event = PropertiesEvent;
+
+    type InitCommand = PropertiesInitCommand;
     type InitEvent = PropertiesInitEvent;
+
     type Error = Error;
 
-    fn init(event: Self::InitEvent) -> Result<Self, Self::Error> {
-        let (handle, version, details) = event.unpack();
-        let krill_version = details.krill_version;
-
-        if version != 0 {
-            Err(Error::custom("version for init event for properties MUST be 0"))
-        } else {
-            Ok(Properties {
-                handle,
-                version: 1, // init for 0 was applied
-                krill_version,
-            })
+    fn init(handle: MyHandle, event: PropertiesInitEvent) -> Self {
+        Properties {
+            handle,
+            version: 1, // init for 0 was applied
+            krill_version: event.krill_version,
         }
     }
 
+    fn process_init_command(command: PropertiesInitCommand) -> Result<Self::InitEvent, Self::Error> {
+        Ok(PropertiesInitEvent {
+            krill_version: command.into_details().krill_version,
+        })
+    }
+
     fn version(&self) -> u64 {
         self.version
     }
 
-    fn apply(&mut self, event: Self::Event) {
+    fn increment_version(&mut self) {
```

Comment:
Should this be a default fn body in the trait, as you seem to do this in a lot of places...

Reply:
It needs to modify the data of the implementing struct, so I am unsure how.